sdk/rust/src/processor/handler.rs

```diff
@@ -0,0 +1,243 @@
+use messages::state_context::TpStateSetRequest;
+use messages::state_context::TpStateSetResponse;
+use messages::state_context::TpStateSetResponse_Status;
+use messages::validator::Message_MessageType;
+
+use messaging::stream::MessageSender;
+use messaging::stream::SendError;
+use messaging::stream::ReceiveError;
+use messaging::zmq_stream::ZmqMessageSender;
+
+#[derive(Debug)]
+pub enum ApplyError {
+    InvalidTransaction(String),
+    InternalError(String)
+}
+
+impl std::error::Error for ApplyError {
+    fn description(&self) -> &str {
+        match *self {
+            ApplyError::InvalidTransaction(ref msg) => msg,
+            ApplyError::InternalError(ref msg) => msg
+        }
+    }
+
+    fn cause(&self) -> Option<&std::error::Error> {
+        match *self {
+            ApplyError::InvalidTransaction(_) => None,
+            ApplyError::InternalError(_) => None
+        }
+    }
+}
+
+impl std::fmt::Display for ApplyError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match *self {
+            ApplyError::InvalidTransaction(ref s) =>
+                write!(f, "InvalidTransaction: {}", s),
+            ApplyError::InternalError(ref s) =>
+                write!(f, "InternalError: {}", s)
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum ContextError {
+    /// Raised for an authorization error
+    AuthorizationError(String),
+    /// Raised when an Unknown error occurs
+    UnknownError(String),
+    /// Raised for a ProtobufError is returned when serialized.
+    SerializationError(Box<StdError>),
+    /// Raise when an error is return when sending a message
+    SendError(Box<StdError>),
+    /// Raise when an error is return when sending a message
+    ReceiveError(Box<StdError>),
+}
+
+impl std::error::Error for ContextError {
+    fn description(&self) -> &str {
+        match *self {
+            ContextError::AuthorizationError(ref msg) => msg,
+            ContextError::UnknownError(ref msg) => msg,
+            ContextError::SerializationError(ref err) => err.description(),
+            ContextError::SendError(ref err) => err.description(),
+            ContextError::ReceiveError(ref err) => err.description(),
+        }
+    }
+
+    fn cause(&self) -> Option<&std::error::Error> {
+        match *self {
+            ContextError::AuthorizationError(_) => None,
+            ContextError::UnknownError(_) => None,
+            ContextError::SerializationError(ref err) => Some(err.borrow()),
+            ContextError::SendError(ref err) => Some(err.borrow()),
+            ContextError::ReceiveError(ref err) => Some(err.borrow()),
+        }
+    }
+}
+
+impl std::fmt::Display for ContextError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match *self {
+            ContextError::AuthorizationError(ref s) =>
+                write!(f, "AuthorizationError: {}", s),
+            ContextError::UnknownError(ref s) =>
```

Comment:
UnknownError seems like it could be abused rather easily. It looks like this is just being used when validating a response from the validator, so maybe something like `ConsistencyError` or `IntegrityError`?